zerver/lib/openapi.py

```diff
@@ -25,28 +20,48 @@
     }
 }
 
+class OpenAPISpec():
+    def __init__(self, path: str) -> None:
+        self.path = path
+
+    def reload(self) -> None:
+        with open(self.path) as f:
+            yaml_parser = YamoleParser(f)
+        self.data = yaml_parser.data
+        self.last_update = os.path.getmtime(self.path)
+
+    def spec(self) -> Dict[str, Any]:
+        """Reload the OpenAPI file if it has been modified after the last time
+        it was read, and then return the parsed data.
+        """
+        last_modified = os.path.getmtime(OPENAPI_SPEC_PATH)
+        if self.last_update is not None and self.last_update < last_modified:
```

Comment:
Might as well use `!=` rather than `<`, in case someone renames an old version back into place.